app/controllers/subscriptions_controller.rb

```diff
@@ -4,7 +4,7 @@ class SubscriptionsController < ApplicationController
   def new
     @origin = origin_param if origin_param&.start_with?(%r{/\w})
     session[:subscription_origin] = @origin
-
+    @point_coordinates = params[:search_criteria] ? Geocoding.new(params[:search_criteria][:location]).coordinates : nil
```

Comment:
Rather than Geocode something twice, why don't we just pass the coordinates through?